git-packages.md

```diff
@@ -0,0 +1,183 @@
+package version.
+
+### Potential issues
+
+XBPS is able to track the versions of dynamic libraries, making it capable of
+performing safe partial upgrades. This, however, is not the case when using
+development packages. Generally speaking, soname versioning is done by the
+upstream project, which probably doesn't care about tracking ABI breaks during
+development, only once a release is made. This means that updating a library to
+a development version can, potentially, break utilities that depend on it, even
+though XBPS happily installed it without incident.
+
+## Patching software
+
+If the prospect of running bleeding edge software sounds too worrying, or a
+package in the official repositories is broken because it lacks a specific patch
+(note that in this case it is recommended to ask upstream for a new release),
+including specific patches is also possible.
+
+Once the package has been built and tested with the patches, the
+[CONTRIBUTING](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md)
+file in void-packages can help you get your changes into the official
+void-packages repository.
+
+### Downloading a patch
+
+For projects hosted on GitHub, a patch file for a specific commit can be found
+in `https://github.com/<user>/<project>/commit/<commit_hash>.patch`, while a
+patch for a Pull Request can be found in
+`https://github.com/<user>/<project>/pull/<pull_number>.patch`. Once these
+patches are downloaded, they should be moved to `srcpkgs/<package>/patches/`.
+All patches in that directory should have the same format, which means each
+section in the patches should start with either:
+
+```
+--- tools/static-nodes.c.orig
++++ tools/static-nodes.c
+```
+
+or
+
+```
+--- a/tools/static-nodes.c.orig
++++ b/tools/static-nodes.c
+```
+
+For the first option, no change is necessary, but for the second option it is
+necessary to add `patch_args="-Np1"` to the template or remove `a/` and `b/`
+from the patch.
+
+### Backporting patches
+
+It is possible that a patch can't be applied automatically to the source tree
+and needs to be fixed. In this case, it can be necessary to create a patch
+yourself. This can be done using git, for example.
+
+Start by cloning the repository and locating the commit(s) containing the fix
+you need. Once you have that information, checkout the released version that is
+currently packaged with
+[git-checkout(1)](https://man.voidlinux.org/git-checkout.1):
+
+```
+$ git checkout <version>
+```
+
+Then, use [git-cherry-pick(1)](https://man.voidlinux.org/git-cherry-pick.1) to
+apply the changes from the commit:
+
+```
+$ git cherry-pick <commit_hash>
+```
+
+You might need to solve some conflicts in the affected files, which are
+annotated as shown below:
+
+```
+<<<<<<< HEAD
+this is some content to mess with
+content to append
+=======
+totally different content to merge later
+>>>>>>> new_branch_to_merge_later
+```
+
```

Comment:
i think it might be useful to explain what the annotations mean, and how one resolves conflicts?

Reply:
Fair enough, I will try to search for a good article to link to, perhaps. Not sure I'd do a great job of it.